### DISCUSSION TITLE
Testing Pipeline

### DISCUSSION THREAD

**Discussion by fweth** [KaTeX/KaTeX#3909] [General] [answered]
Hello, I am motivated to resolve [this issue](https://github.com/KaTeX/KaTeX/issues/1233), but I am not very familiar with testing pipelines. Therefore, I have two questions:

1.   Can I run the screenshotter after forking the repository to test it before creating a pull request? Or is this not possible due to secrets for BrowserStack?
2.   Would it be feasible to host the contents of [ss_data.yaml](https://github.com/KaTeX/KaTeX/blob/main/test/screenshotter/ss_data.yaml) on a website, to facilitate quick visual testing?

**Comment by edemaine** [comment, marked as answer]
1. You can run the screenshotter manually for Chrome and Firefox, without BrowserStack, using docker as documented in `KaTeX/dockers/screenshotter/README.md`. This doesn't include Safari without BrowserStack credentials, but hopefully that's OK for testing purposes.
2. I'm not sure what you mean. Are you asking us to host it on a website? Or are you asking about creating one?

P.S. Mainly I think #3133 needs review. If you could provide input on whether you think the changes are good/safe, that would help.

**Reply by fweth** [reply]
Thank you! With the second part, I just meant that I had a hard time converting the YAML file, it would be nice to have a bunch of testing input either as HTML or as text with the default delimiters so that I can copy paste it into an HTML file. Just to quickly get a bunch of equations on the screen.

**Reply by fweth** [reply]
Do you know if this would also work in the cloud?  I didn't manage to install Docker on my local machine, but I tried to run the test in [GitPod](https://gitpod.io/) (where Docker is installed) and got:

```
Container 77ee3e214c29 started, creating screenshots...
Type Error: Cannot read properties of undefined (reading '/workspace/KaTeX/.pnp.cjs')
```

What could be the problem?
